### PR TITLE
Move Camera and Things

### PR DESCRIPTION
**Pull request by pbardea**
Address #18 and #14 

Turns out with open GL you don't want to write you're own vector rotating math.

USE THE LIBRARY LUKE!

So, even though I've been coming back to the math for the past couple of weeks, in the end it took like half an hour once I realized that the libraries can do most of the heavy lifting.

The code is currently ugly, to make it work. Will refactor later tonight, but I thought you guys may want to play with it :D Would make some neat GIFs for tomorrow!

### CODE REVIEW

File: src/Scene.cpp [hunk 1]
```diff
@@ -38,26 +39,56 @@ const std::vector<PointLight>& Scene::getPointLights() const {
...
   setView();
 }
 
 void Scene::moveCameraRight() {
-  cameraPos.x -= 0.05;
-  cameraTarget.x -= 0.05;
+  float PI_BY_2 = glm::pi<float>() / 2.0;
+  glm::vec3 dir = glm::normalize(cameraTarget - cameraPos);
+  dir = glm::rotateY(dir, PI_BY_2);
+  dir *= 0.05;
```
Comment (AndrewMcBurney) [inline src/Scene.cpp]: Y'all mind if you do this in one line?

```cpp
dir = glm::rotateY(dir, PI_BY_2);
dir *= 0.05;
```
=>
```cpp
dir = glm::rotateY(dir, PI_BY_2) * 0.05;
```

File: src/Scene.cpp [hunk 2]
```diff
@@ -38,26 +39,56 @@ const std::vector<PointLight>& Scene::getPointLights() const {
 // Camera movement
 
 void Scene::moveCameraLeft() {
-  cameraPos.x += 0.05;
-  cameraTarget.x += 0.05;
+  float PI_BY_2 = glm::pi<float>() / 2.0;
+  glm::vec3 dir = glm::normalize(cameraTarget - cameraPos);
+  dir = glm::rotateY(dir, PI_BY_2);
```
Comment (AndrewMcBurney) [inline src/Scene.cpp]: Y'all mind if you do this in one line?

```cpp
dir = glm::rotateY(dir, PI_BY_2);
dir *= 0.05;
```
=>
```cpp
dir = glm::rotateY(dir, PI_BY_2) * 0.05;
```

File: src/Scene.cpp [hunk 3]
```diff
@@ -99,3 +97,4 @@ void Scene::setView() {
 }
 
 const glm::vec3 Scene::upVector(0, 1, 0);
+const float Scene::stepSize = 0.05;
```
Comment (AndrewMcBurney) [inline src/Scene.cpp]: nice 💯

File: src/Scene.cpp [hunk 4]
```diff
@@ -38,26 +39,54 @@ const std::vector<PointLight>& Scene::getPointLights() const {
 // Camera movement
 
 void Scene::moveCameraLeft() {
-  cameraPos.x += 0.05;
-  cameraTarget.x += 0.05;
+  float PI_BY_2 = glm::pi<float>() / 2.0;
+  glm::vec3 dir = glm::normalize(cameraTarget - cameraPos);
+  dir = glm::rotateY(dir, PI_BY_2) * stepSize;
+  cameraTarget += dir;
```
Comment (davepagurek) [inline src/Scene.cpp]: I think we may want to just update the x and z positions, if we're rotating about y

**Pull request merged**